### ISSUE TITLE
Bug: space-unary-ops is suddenly flagging spaces it shouldn't be

### ISSUE THREAD

**Issue by Twipped** [eslint/eslint#17685]
### Environment

Node version: v18.18.0
npm version: v9.4.2
Local ESLint version: v8.52.0 (Currently used)
Global ESLint version: Not found
Operating System: linux 5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko

### What parser are you using?

Default (Espree)

### What did you do?

<details>
<summary>Configuration</summary>

```
{
  "env": {
    "es2022": true
  },
  "rules": {
    "space-unary-ops": [
      "error",
      {
        "words": true,
        "nonwords": false
      }
    ]
  }
}
```
</details>

```js
const a = new Error();
```

### What did you expect to happen?

Lint should pass

### What actually happened?

Something has changed in the last two days. I did an npm update a couple hours ago and eslint started flagging dozens of spaces following `new`. There has been no changes in our configuration. I rolled back package-lock and the problem went away.

### Link to Minimal Reproducible Example

https://stackblitz.com/edit/stackblitz-starters-msp9mu?file=.eslintrc.json

### Participation

- [ ] I am willing to submit a pull request for this issue.

### Additional comments

This appears to be [caused by](https://github.com/acornjs/acorn/issues/1252) the recent update to `acorn` 8.11.0. When I force it to v8.10.0, the problem goes away. 

I went ahead and opened this issue because I figured others would come here trying to figure out what's up.

**Comment by mdjermanovic** [comment]
Closing as a duplicate of https://github.com/eslint/eslint/issues/17684 so that we have a single issue where we track this.
